rust/chains/hyperlane-cosmos/src/payloads/aggregate_ism.rs

```diff
@@ -1,17 +1,30 @@
+use hyperlane_core::{HyperlaneMessage, RawHyperlaneMessage};
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct ModulesAndThresholdRequest {
-    pub modules_and_threshold: ModulesAndThresholdRequestInner,
+    modules_and_threshold: ModulesAndThresholdRequestInner,
+}
+
+impl ModulesAndThresholdRequest {
+    pub fn new(message: &HyperlaneMessage) -> Self {
+        Self {
+            modules_and_threshold: ModulesAndThresholdRequestInner {
+                message: hex::encode(RawHyperlaneMessage::from(message)),
+            },
+        }
+    }
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-pub struct ModulesAndThresholdRequestInner {
-    pub message: String, // hexbinary
+struct ModulesAndThresholdRequestInner {
+    /// Hex-encoded Hyperlane message
+    pub message: String,
```

Comment:
one of my comments in the big PR is to stop using these strings entirely, curious to hear what you think

Reply:
yeah I like that